examples/custom-assessments/src/main.py

```diff
@@ -0,0 +1,233 @@
+        """
+        # Create a dictionary for the results. Prepend "state_" to state columns.
+        df = pd.DataFrame(states).add_prefix("state_")
+
+        # Add action columns
+        df = pd.concat([df, pd.DataFrame(actions).add_prefix("action_")], axis=1)
+
+        # Add initial condition columns
+        # Check the type of the initial_condition argument
+        if isinstance(self.initial_condition, dict):  # If it is a dictionary
+            # Use the dictionary keys and values to create new columns
+            df[[f"config_{k}" for k in self.initial_condition.keys()]] = list(
+                self.initial_condition.values()
+            )
+        elif isinstance(self.initial_condition, list):  # If it is a list
+            # Use the list index and values to create new columns
+            df[
+                [f"config_{i}" for i in range(len(self.initial_condition))]
+            ] = self.initial_condition
+        else:  # If it is neither a list nor a dictionary
+            # Raise an exception
+            raise TypeError("The config argument must be a list or a dictionary")
+
+        # Add reward, terminated, truncated, episode id, and step id columns
+        df["reward"] = pd.Series(rewards)
+        df["reward_cumsum"] = df["reward"].cumsum()
+        df["terminated"] = pd.Series(terminated)
+        df["truncated"] = pd.Series(truncated)
+        df["episode_id"] = self.episode_id
+        df["step_id"] = pd.Series(np.arange(n_steps + 2, dtype=int))
+
+        return df
+
+    def run_episode(
+        self,
+        initial_condition: Dict[str, int],
+        episode_id: int = 0,
+        max_steps: int = 100,
+    ) -> pd.DataFrame:
+        """Run an episode for the initial conditions using the agent policy and the
+        simulator.
+
+        Args:
+            initial_condition (dict): The initial condition for the simulation.
+            episode_id (int, optional): The id of the episode. Defaults to 0.
+            max_steps (int, optional): The maximum number of steps for the
+                episode. Defaults to 100.
+
+        Returns:
+            pd.DataFrame: The dataframe containing the results of the episode run.
+        """
+        # assign values to object properties
+        self.initial_condition = initial_condition
+        self.episode_id = episode_id
+        self.max_steps = max_steps
+
+        # Reset state with initial condition and return observable state
+        observable_state, _ = self.sim.reset(options=self.initial_condition)
+
+        # Initialize lists for episode logging
+        states = []
+        actions = []
+        rewards = []
+        terminated = []
+        truncated = []
+        # Test for max_steps
+        for step in range(self.max_steps):
+            # Get full current state and save it
+            full_state = (
+                self.sim.state.copy()
+                if type(self.sim.state) == dict
+                else self.sim.state
+            )
```

Comment:
Alright, I think I left comments to an older version of the code.
I'm not sure if you missed it but I wanted to suggest using this approach:

```python
import copy
...
full_state = copy.deepcopy(self.sim.state)
```
which I think should work with any type

Reply:
Thanks. I made the change.